pkg/controller/registry/reconciler/grpc.go

```diff
@@ -198,12 +199,18 @@ func (c *GrpcRegistryReconciler) ensureUpdatePod(source grpcCatalogSourceDecorat
 			updatePod.Labels[CatalogSourceUpdateKey] = ""
 
 			// Update the update pod to promote it to serving pod
-			_, err := c.OpClient.KubernetesInterface().CoreV1().Pods(source.GetNamespace()).Update(context.TODO(), updatePod, metav1.UpdateOptions{})
-			if err != nil {
-				return errors.Wrapf(err, "error creating new pod: %s", source.Pod().GetName())
+			forceApply := true
+			opts := metav1.PatchOptions{
+				Force:        &forceApply,
+				FieldManager: "operator-lifecycle-manager-registry-reconciler",
 			}
 
-			break
+			updatePodBytes, err := client.Apply.Data(updatePod)
+			if err != nil {
+				return errors.Wrapf(err, "error updating catalog source pod: %s", source.Pod().GetName())
```

Comment:
can we make this error log more specific like "error serializing updated catalog source pod to bytes: ..."

The wrapped error will have some more info, but I wouldn't want this error to look too similar to the one below.